libs/cart/src/drivers/magento/transforms/outputs/cart-item.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+
+import { MagentoCartItem } from '../../models/outputs/cart-item';
+import { DaffCartItem } from '../../../../models/cart-item';
+
+/**
+ * Transforms magento carts into an object usable by daffodil.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoCartItemTransformer {
+
+  /**
+   * Transforms the magento MagentoCartItem from the magento cart query into a DaffCartItem.
+   * @param response the response from a magento cart query.
+   */
+  transform(cartItem: MagentoCartItem): DaffCartItem {
+    return {
+      ...cartItem,
```

Comment:
I see this in a lot of places, and I don't think we want this. Won't this just put a bunch of garbage on the DaffCartItem on state? Because we shouldn't assume that the magento and daffodil models will be the same.

Reply:
This is to keep the transformers covariant as per [this discussion](https://github.com/graycoreio/daffodil/pull/654#discussion_r385334319) with Damien.

Reply:
That make negative sense to me. I'll ask him about it